packages/cli/src/tools/windows/androidWinHelpers.ts

```diff
@@ -0,0 +1,301 @@
+ */
+export const getAndroidSdkRootInstallation = () => {
+  // TODO: Check if the path exists?
+  const env = process.env.ANDROID_SDK_ROOT || process.env.ANDROID_HOME;
+  const installPath = env
+    ? // Happens if previous installations or not fully completed
+      env
+    : // All Android zip files have a root folder, using `Android` as the common place
+      join(getUserAndroidPath(), 'Sdk');
+
+  return installPath;
+};
+
+/**
+ * Installs an Android component (e.g.: `platform-tools`, `emulator`)
+ * using the `sdkmanager` tool and automatically accepting the licenses.
+ */
+export const installComponent = (component: string, androidSdkRoot: string) => {
+  return new Promise((done, error) => {
+    const sdkmanager = join(androidSdkRoot, 'tools', 'bin', 'sdkmanager.bat');
+
+    const command = `"${sdkmanager}" --sdk_root="${androidSdkRoot}" "${component}"`;
+
+    const child = executeCommand(command);
+    let stderr = '';
+
+    child.stdout.on('data', data => {
+      if (data.includes('(y/N)')) {
+        child.stdin.write('y\n');
+      }
+    });
+
+    child.stderr.on('data', data => {
+      stderr += data.toString('utf-8');
+    });
+
+    child.on('close', data => {
+      if (data === 0) {
+        done();
+      } else {
+        error({stderr});
+      }
+    });
+    child.on('error', error);
+  });
+};
+
+/**
+ * For the given custom Hypervisor and the output of `emulator-check accel`
+ * returns the preferred Hypervisor to use and its installation status.
+ * The recommendation order is:
+ * 1. WHPX
+ * 2. HAXM if Intel
+ * 3. AMDH if AMD
+ */
+const parseHypervisor = (
+  status: string,
+  customHypervisor: 'HAXM' | 'AMDH',
+): HypervisorStatus | null => {
+  /**
+   * Messages:
+   * Android Emulator requires an Intel processor with VT-x and NX support.  Your CPU: 'AuthenticAMD'
+   * HAXM is not installed, but Windows Hypervisor Platform is available.
+   * WHPX (10.0.19041) is installed and usable.
+   * * This message outputs for WHPX and when the AMD Hypervisor is installed
+   * HAXM version 6.2.1 (4) is installed and usable.
+   * HAXM is not installed on this machine
+   */
+
+  if (
+    status.includes(
+      'is not installed, but Windows Hypervisor Platform is available.',
+    )
+  ) {
+    return {
+      hypervisor: 'WHPX',
+      installed: false,
+    };
+  }
+
+  if (/WHPX \((\d|\.)+\) is installed and usable\./.test(status)) {
+    return {
+      hypervisor: 'WHPX',
+      installed: true,
+    };
+  }
+
+  if (/is installed and usable\./.test(status)) {
+    return {
+      hypervisor: customHypervisor,
+      installed: true,
+    };
+  }
+
+  if (status.includes("Your CPU: 'AuthenticAMD'")) {
+    return {
+      hypervisor: customHypervisor,
+      installed: false,
+    };
+  }
+
+  if (status.includes('is not installed on this machine')) {
+    return {
+      hypervisor: 'none',
+      installed: false,
+    };
+  }
+
+  return null;
```

Comment:
Is this a valid expected outcome? If not, then it might be better to throw so we get reports from people when things change (especially as parsing human readable output can be so flaky).

Reply:
Yes it's valid and handled [a bit later](https://github.com/react-native-community/cli/pull/1109/files#diff-a0a0172a96292d7e8a32ce5f2822db63R240)